### PR TITLE
Tests for computed namespace constructor

### PR DESCRIPTION
**Pull request by adamretter**

### CODE REVIEW

**Comment by duncdrum** [comment]
[L96](https://github.com/eXist-db/exist/pull/3781/checks?check_run_id=2159559099#step:5:96):

```
Error:  Failed to execute goal com.mycila:license-maven-plugin:4.0.rc1:check (check-headers) on project exist-core: Some files do not have the expected license header -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <args> -rf :exist-core
Error: Process completed with exit code 1.
```

**Review by duncdrum — CHANGES_REQUESTED** [review]
does `exist-core/src/test/xquery/xquery3/computed-namespace-constructor.xql` need a license?

**Comment by adamretter** [comment]
@duncdrum Oops! I added that now.

**Review by duncdrum — APPROVED** [review]
CI has network issues, but all in all this looks good

**Review by dizzzz — APPROVED** [review]

**Comment by dizzzz** [comment]
OK to pull?

File: exist-core/src/main/java/org/exist/xquery/NamespaceConstructor.java [hunk 1]
```diff
@@ -110,9 +107,9 @@ public Sequence eval(Sequence contextSequence, Item contextItem) throws XPathExc
             throw new XPathException(this, ErrorCodes.XQDY0102, "Cannot override already defined ns");
         }
 
-        if (prefix.equals("xmlns")) {
+        if (prefix.equals(XMLConstants.XMLNS_ATTRIBUTE)) {
```
Comment (line-o) [inline exist-core/src/main/java/org/exist/xquery/NamespaceConstructor.java:110]: For me this is refactoring code, not reformatting it.

File: exist-core/src/main/java/org/exist/xquery/NamespaceConstructor.java [hunk 2]
```diff
@@ -34,19 +33,19 @@
...
+ *
  * @author wolf
  */
 public class NamespaceConstructor extends NodeConstructor {
 
     private Expression qnameExpr;
     private Expression content = null;
 
-    public NamespaceConstructor(XQueryContext context) {
+    public NamespaceConstructor(final XQueryContext context) {
```
Comment (line-o) [inline exist-core/src/main/java/org/exist/xquery/NamespaceConstructor.java:44]: If this is just reformatting code is debatable.

**Comment by line-o** [comment]
@dizzzz I do not know about appveyor. It seems to always fail on ubuntu, but you could try to restart it once again. I am still in favour of dropping appveyor completely.

The code analysis fails because of the code formatting _and_ the minor refactoring that happened in NamespaceConstructor.java - that triggers new analysis and the eval method had and still has a _high_ complexity.

**Comment by line-o** [comment]
So, I would like to see the commit message change to `[ignore] refactor and reformat NamespaceConstructor` other than that it looks good to me.

**Comment by adamretter** [comment]
@line-o I changed the commit message as requested

**Pull request merged**